components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -481,6 +508,91 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
+  if ([self isMobileLandscape]) {
+    self.contentViewController.preferredContentSize = newPreferredContentSize;
+    return;
+  }
+  CGPoint contentOffset =
+      [self calculateContentOffsetWithPreferredContentHeight:preferredContentHeight];
+  CGFloat precentageOfFullScreen =
+      [self precentageOfFullScreenWithPreferredContentHeight:preferredContentHeight];
+  BOOL shouldSetContentHeight = NO;
```

Comment:
do we not set the content height if it's set to smaller than before? can you provide a comment to explain a bit of the logic?